derive/src/lib.rs

```diff
@@ -56,6 +59,26 @@ use syn::{
     Variant,
 };
 
+use once_cell::sync::Lazy;
+
+struct IdentHelper(Lazy<String>);
+
+impl quote::ToTokens for IdentHelper {
+    fn to_tokens(&self, tokens: &mut TokenStream2) {
+        Ident::new(&self.0, Span::call_site()).to_tokens(tokens)
+    }
+}
+/// Get the name of the scale-info crate, to be robust against renamed dependencies.
+static SCALE_INFO: IdentHelper = IdentHelper(Lazy::new(|| {
```

Comment:
I have a (probably irrational) aversion to global singletons unless absolutely necessary, though there is nothing wrong with them per se they are often a code smell.

Reply:
Conceptually these aren't really globals of the bad kind though; they are constants and not really "state" (and obviously not mutable state). 
Imagine a parallel universe where an `Ident` as actually a `&'static str` and `proc-macro-crate::crate_name` was a `const fn`, wouldn't it be more natural to have `const SCALE_INFO: Ident = proc-macro-crate::crate_name("scale-info").expect("wut?!");`? The crate name used is a value that does not vary, it's constant.

Reply:
I understand your point, but like I said it's irrational :grin: